### PR TITLE
[AN-257] Update mysql version to 5.7

### PR DESCRIPTION
**Pull request by LizBaldo**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/AN-257

### What

- Points the local leo mysql database to use 5.7 for both local runs, as well as unit testing via GHA
- Slight changes to some liquibase migration files to adapt to the mysql 5.6 -> 5.7 breaking changes. Note that I added the checksum validations for each table as to not break dev and staging like #4802 did 🤦‍♀️ 
- Sets the option to allow for public key retrieval, which is no longer the default with mysql 8.0, see https://customerportal.qlik.com/article/MySQL-JDBC-connection-Public-Key-Retrieval-is-not-allowed

### Why

-

### What to test

- Testing on a BEE with preexisting data from dev

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

**Review by lucymcnatt — APPROVED** [review]
Thanks for adding the comments!

File: .github/workflows/unit_test.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
 
     services:
       mysql:
-        image: mysql/mysql-server:5.6
+        image: mysql/mysql-server:8.0
```
Comment (aednichols) [inline .github/workflows/unit_test.yml]: `mysql/mysql-server` seems to be abandoned, no [tags](https://hub.docker.com/r/mysql/mysql-server/tags) have been updated in 2 years.

We should consider moving to just `mysql`, which expands to `library/mysql` and is the official [image](https://hub.docker.com/_/mysql).

File: http/src/test/resources/reference.conf [hunk 2]
```diff
@@ -37,7 +37,7 @@ mysql {
   port = 3311
   db {
     driver = "com.mysql.cj.jdbc.Driver"
-    url = "jdbc:mysql://"${mysql.host}":"${mysql.port}"/leotestdb?createDatabaseIfNotExist=true&useSSL=false&rewriteBatchedStatements=true&nullNamePatternMatchesAll=true"
+    url = "jdbc:mysql://"${mysql.host}":"${mysql.port}"/leotestdb?createDatabaseIfNotExist=true&allowPublicKeyRetrieval=true&useSSL=false&rewriteBatchedStatements=true&nullNamePatternMatchesAll=true"
```
Comment (aednichols) [inline http/src/test/resources/reference.conf:40]: I would love to enable SSL everywhere and disable `allowPublicKeyRetrieval` but that seems like a project for another PR.

**Review by aednichols — APPROVED** [review]
We should consider MySQL 8.4, which is the current LTS [0].

It buys us [a lot more time](https://cloud.google.com/sql/docs/mysql/db-versions#database-version-support) than 8.0, which expires in July 2026; meaning it's only a 1.5 year extension over 5.6.

[0] The `mysql:lts` tag currently points to `mysql:8.4`

File: http/src/test/resources/reference.conf (hunk 2 above)
Comment (LizBaldo) [reply http/src/test/resources/reference.conf:40]: Yeah, I checked that this is only used for our unit testing at least

File: .github/workflows/unit_test.yml (hunk 1 above)
Comment (LizBaldo) [reply .github/workflows/unit_test.yml]: Good call, let me try that

**Comment by LizBaldo** [comment]
Haha lots of folks mad about this originally undocumented breaking change with mySQL 8.4: https://bugs.mysql.com/bug.php?id=114838

**Comment by LizBaldo** [comment]
Grrr, driving me a bit nuts, the FK is referencing the primary key, how can it not be unique already? 🙄 
`Failed to add the foreign key constraint. Missing unique key for constraint 'FK_CLUSTER_ID' in the referenced table 'CLUSTER'`

**Pull request merged**